### DISCUSSION TITLE
Invite only for Course enrollment

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Lern/Community#14] [Query]
Hi @amit-tarento

Hope you are doing well!..

**Site Environments Details** - SunbirdEd portal of version 4.4.1

We are looking for an 'invite-only' feature for the course enrollment which could be set up while creating a batch.
This is to restrict enrollment to specific users instead of publicly open like now.

This seems to be deprecated in the latest version. Could you help us with what will it require to get this back or any available alternative?

**Note -** we are trying to use create batch API with the 'enrollmentType': 'invite-only' value but the response of that API is 

```
{
    "id": "api.course.batch.create",
    "ver": "v1",
    "ts": "2022-06-14 07:14:27:890+0000",
    "params": {
        "resmsgid": null,
        "msgid": "f1966c89-f84d-4f4a-9843-924e07f7a8f1",
        "err": "INVALID_PARAMETER_VALUE",
        "status": "INVALID_PARAMETER_VALUE",
        "errmsg": "Invalid value invite-only for parameter enrollmentType. Please provide a valid value."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}
```

cc @coolbung, @Renuka-S, @vinod303

**Comment by pramodkvarma** [comment]
This is definitely an important feature for SB-Lern to have (although DIKSHA may not need it). I remember SB-CB team implementing it. I think they used visibility attribute of collection object to achieve this. 

@vrayulu do you know the details? Can you suggest?

@komalm if you are implementing this enhancement make sure to contribute back please.

**Reply by amitec006** [reply]
@komalm We have enabled this feature back in sunbird with release-4.9.0.

**Reply by Renuka-S** [reply]
@amit-tarento Can you please specify what all need to be upgraded to enable this? Also please share if there is any PR for this patch. cc: @coolbung @komalm

**Reply by amitec006** [reply]
@Renuka-S Please check the below PR for the invite-only changes for course creation.
https://github.com/project-sunbird/sunbird-course-service/pull/443

**Reply by pranavirabatti** [reply]
@amit-tarento you have added these changes in the backend to get this feature back. Need to develop its UI? Looks like changes are not made to the frontend.

![image (2)](https://user-images.githubusercontent.com/22250006/175304902-b909e5b9-4810-43de-ae5c-7430655c27c2.png)

**Reply by reshmi-nair** [reply]
@pranavirabatti Front-end change are handled by SunbirdEd Building Block, so please reach out in https://github.com/orgs/Sunbird-Ed/discussions.

**Comment by Xitija** [comment]
Hello @amit-tarento sir,
Can you please provide the api for creating the invite-only batches .
I want to enable this feature through the frontend.
It will be really helpful for me.

**Reply by vinukumar-vs** [reply]
@reshmi-nair @Hari-stackroute Can you please the details

**Reply by reshmi-nair** [reply]
@Xitija Pls refer to this documentation: https://lern.sunbird.org/learn/product-and-developer-guide/batch-service/api-documentation

Create Batch API can be used to create batch with invite-only option. enrollmentType field in the request should have value as "invite-only"
